### PR TITLE
SHIRO-570: Only accept a cookie value when the request uses the proper path

### PR DESCRIPTION
**Pull request by ankon**

### CODE REVIEW

File: web/src/main/java/org/apache/shiro/web/servlet/SimpleCookie.java [hunk 1]
```diff
@@ -362,8 +362,14 @@ public String readValue(HttpServletRequest request, HttpServletResponse ignored)
         String value = null;
         javax.servlet.http.Cookie cookie = getCookie(request, name);
         if (cookie != null) {
-            value = cookie.getValue();
-            log.debug("Found '{}' cookie value [{}]", name, value);
+            // Validate that the cookie is used at the correct place.
+            String path = StringUtils.clean(getPath());
+            if (path != null && !request.getRequestURI().startsWith(path)) {
```
Comment (bdemers) [inline web/src/main/java/org/apache/shiro/web/servlet/SimpleCookie.java]: Should we just check if `getPath()` != `calcualtePath(...)` ?

Comment (ankon) [inline web/src/main/java/org/apache/shiro/web/servlet/SimpleCookie.java]: I might be missing something, but if `getPath()` returns a non-null/non-empty string, then `calculatePath(request)` would return that value, which would trivially be equal?

**Comment by ankon** [comment]
While looking at the question about `#calculatePath()`, I checked the rules for path matching for browsers, and found an edge case: the logic would accept a prefix that is not a complete path, for example it would accept requests from '/foobar/something.jsp' as matching for the cookie with a path of '/foo'.

**Comment by bdemers** [comment]
+1 🍪 !

**Pull request closed without merging**